### PR TITLE
[RDY] Return hooks, allowing the client to call them instead of the backend.

### PR DESCRIPTION
**Pull request by Hjdskes**
As the title says. (After #24)

~~Right now I have to cherry-pick 4fa1a5c3f286ede7ea58972043c6bee611070948 so that Guice does not need to instantiate the hooks anymore. Currently, these hooks get an `IDisplay` injected, however the `EclipseDisplay` implementation cannot be instantiated at that time because it needs a `Composite`. Having the commands return the "hooks" instead solves this problem, in addition to resolving the code duplication of having to schedule all hooks on the UI thread as noted by Skip (https://github.com/spoofax-shell/spoofax-shell/pull/24#discussion_r65813358).~~

### CODE REVIEW

File: org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/client/IRepl.java [hunk 1]
```diff
@@ -22,17 +23,20 @@
...
      * @throws CommandNotFoundException
      *             When the command could not be found.
      */
-    default void eval(String input) throws MetaborgException, CommandNotFoundException {
+    // TODO: return Optional?
+    default IHook eval(String input) throws MetaborgException, CommandNotFoundException {
         input = input.trim();
         if (input.length() == 0) {
-            return;
+            return null;
```
Comment (Hjdskes) [inline org.metaborg.spoofax.shell.core/src/main/java/org/metaborg/spoofax/shell/client/IRepl.java]: I still have to update the clients to handle this. Perhaps it'd be better to return an `Optional` instead.

**Comment by Hjdskes** [comment]
The WIP "Run in REPL" commit currently breaks the REPL in that starting it, closing it and starting it again does not seem to reload the spoofax commands.

**Comment by Hjdskes** [comment]
The previous comment has been resolved in 0b4a8d48630bf33ad301a33574ebd09861687418.

I believe this is ready for review now. @Balletie @gfokkema

**Comment by gfokkema** [comment]
The REPL does start, however, the interpreted languages aren't available on the classpath.
Therefore no languages work.

**Comment by Hjdskes** [comment]
I removed the Run in REPL functionality (`run-in-repl2`) so that this branch can be merged regardless.

**Pull request merged**